File: program/WordCount/Proc/ruby/Top10WordList.rb
arr=Array.new()
$stdin.each_line {|line|
    count,word=line.split(" ")
    arr.push [count.to_i,word]
}
arr.sort!{|a,b|(b[0]<=>a[0]).nonzero? || a[1]<=>b[1]}
(0..[10,arr.length-1].min-1).each {|i|
    puts "#{arr[i][0]} #{arr[i][1]}"
}

File: program/WordCount/Proc/perl/Top10WordList.pl
while (<>) {
    chomp;
    push @words,[split " "];
}
@words=sort {$b->[0]<=>$a->[0] or $a->[1] cmp $b->[1]} @words;
$count=@words;
for ($i=0;$i<$count && $i<10;$i++) {
    printf("%d %s\n",$words[$i]->[0],$words[$i]->[1]);
}
